tools/api-client/python/lib/bmutils.py

```diff
@@ -98,10 +100,40 @@ def wrap_create_game(self, pbutton, obutton='', opponent=''):
     return retval.data
 
   def wrap_load_game_data(self, game):
-    retval = self.load_game_data(game)
-    if not retval.status == 'ok':
-      raise ValueError, "Failed to call loadGameData, got: " + retval
-    data = retval.data
+    # if we're using a cache directory
+    if self.cachedir:
+      # if the cache directory doesn't exist, create it
+      if not os.path.isdir(self.cachedir):
+        os.makedirs(self.cachedir)
+      # set the path to the cache file for this game
+      cachefile = self.cachedir + "/" + unicode(game) + ".json"
+      # if the cache already has a file for this game
+      if os.path.isfile(cachefile):
+        # get the game data from the cache
+        with open(cachefile) as cache_fh:
+          data = json.load(cache_fh)
+      # otherwise (the cache didn't already have a file for this game)
+      else:
+        # load the game
+        retval = self.load_game_data(game)
+        # if that didn't work, raise an exception
+        if not retval.status == 'ok':
+          raise ValueError, "Failed to call loadGameData, got: " + retval
```

Comment:
While you're touching this code, want to fix the bug in which this blows up because retval is an object, not a string?  I'm pretty sure the fix is just `s/retval/retval.message/` - you can change that here and in the non-cache case below.